samples/OcelotServiceDiscovery/ApiGateway/ocelot.json

```diff
@@ -0,0 +1,19 @@
+﻿{
+  "Routes": [
+    {
+      "DownstreamPathTemplate": "/api/Categories",
+      "DownstreamScheme": "http",
+      "UpstreamPathTemplate": "/Categories",
+      "ServiceName": "downstream-service",
+      "UpstreamHttpMethod": [ "Get" ],
+      "FileCacheOptions": { "TtlSeconds": 15 }
+    }
+  ],
+  "GlobalConfiguration": {
+    "RequestIdKey": "OcRequestId",
+    "AdministrationPath": "/administration",
+    "ServiceDiscoveryProvider": { 
+      "Type": "MyServiceDiscoveryProvider" 
```

Comment:
Please note, this **Type** setting is important for default `ServiceDiscoveryProviderFactory` class which calls mentioned FinderDelegate.
For the second option/way/approach of development we need not this setting at all.
We can leave this setting in config because we cover both ways of development.